src/wallet.js

```diff
@@ -337,14 +345,22 @@ const wallet = {
         } else {
           // If it's the last page, we update the storage and call the next loadAddress (if needed)
           // This is all done on updateHistoryData
+          let oldStore = _.clone(storage.store);
```

Comment:
Why do we have to get it cloned?

Reply:
I save the `oldStore` to set it back after the actions are done. We don't know the `store` that will be implemented for each case, so I clone it to prevent an error if the store is passed as reference, to guarantee when I set it back later it will really be the old store.

Reply:
As discussed privately, we will merge this and handle the `clone` problem in another PR